nordic-nrf51822-armcc/CMake/toolchain.cmake

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if(TARGET_NORDIC_NRF51822_ARMCC_TOOLCHAIN_INCLUDED)
+    return()
+endif()
+set(TARGET_NORDIC_NRF51822_ARMCC_TOOLCHAIN_INCLUDED 1)
+
+# If the memory size has not been defined then default to 16K and print a warning
+# if both 16K and 32K have been defined then abort the build process
+if(NOT YOTTA_CFG_CHIP_NRF51822_16K AND NOT YOTTA_CFG_CHIP_NRF51822_32K)
```

Comment:
I have my doubts about how to use the definitions for the RAM size. My original idea was to make a yotta config like this:
```
"hardware": {
    "ram_size": "32K"
}
```
that would translate into `YOTTA_CFG_HARDWARE_RAM_SIZE` and could easily be used in the toolchain.cmake and remove the need for this "complex" block of cmake code. The reason why I used the `YOTTA_CFG_CHIP_NRF51822_16K` and `YOTTA_CFG_CHIP_NRF51822_32K` definitions is because they already existed, but I would prefer to remove them (if possible).